features/step_definitions/request.rb

```diff
@@ -135,7 +139,16 @@ def build_given(api_version, operation)
 When('the request is sent') do
   params = @api_method.parameters.select { |p| p[0] == :req }.map { |p| opts.delete(p[1]) }
   undo_builder = build_undo_for @api_method.name.to_s.chomp('_with_http_info')  # fail early on missing undo method
-  @response = @api_method.call(*params, opts)
+
+  begin
+    @response = @api_method.call(*params, opts)
+  rescue api_error => e
+    # If we have an exception, make a stub response object to use for assertions
+    # Instead of finding the response class of the method, we use the fact that all
+    # responses returned have the `1` element set to the response code
+    @response = [nil, e.code, nil]
```

Comment:
What does that do if a legit test raises an exception ? How does it look like ?

Reply:
The response that the `assert` methods get back would have a status code that would be 4xx or 5xx
So the assertion would fail since the legit test would say `Then response is 200`